crates/uv-platform-tags/src/abi.rs

```diff
@@ -0,0 +1,447 @@
+
+/// A tag to represent the ABI compatibility of a Python distribution.
+///
+/// This is the second segment in the wheel filename, following the language tag. For example,
+/// in `cp39-none-manylinux_2_24_x86_64.whl`, the ABI tag is `none`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub enum AbiTag {
+    /// Ex) `none`
+    None,
+    /// Ex) `abi3`
+    Abi3,
+    /// Ex) `cp39m`, `cp310t`
+    CPython {
+        gil_disabled: bool,
+        python_version: (u8, u8),
+    },
+    /// Ex) `pypy39_pp73`
+    PyPy {
+        python_version: (u8, u8),
+        implementation_version: (u8, u8),
+    },
+    /// Ex) `graalpy310_graalpy240_310_native`
+    GraalPy {
+        python_version: (u8, u8),
+        implementation_version: (u8, u8),
+    },
+    /// Ex) `pyston38-pyston_23`
+    Pyston {
+        python_version: (u8, u8),
+        implementation_version: (u8, u8),
+    },
+}
+
+impl std::fmt::Display for AbiTag {
+    /// Format an [`AbiTag`] as a string.
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::None => write!(f, "none"),
+            Self::Abi3 => write!(f, "abi3"),
+            Self::CPython {
+                gil_disabled,
+                python_version: (major, minor),
+            } => {
+                if *minor <= 7 {
+                    write!(f, "cp{major}{minor}m")
+                } else if *gil_disabled {
+                    // https://peps.python.org/pep-0703/#build-configuration-changes
+                    // Python 3.13+ only, but it makes more sense to just rely on the sysconfig var.
+                    write!(f, "cp{major}{minor}t")
+                } else {
+                    write!(f, "cp{major}{minor}")
+                }
+            }
+            Self::PyPy {
+                python_version: (py_major, py_minor),
+                implementation_version: (impl_major, impl_minor),
+            } => {
+                write!(f, "pypy{py_major}{py_minor}_pp{impl_major}{impl_minor}")
+            }
+            Self::GraalPy {
+                python_version: (py_major, py_minor),
+                implementation_version: (impl_major, impl_minor),
+            } => {
+                write!(
+                    f,
+                    "graalpy{py_major}{py_minor}_graalpy{impl_major}{impl_minor}_{py_major}{py_minor}native"
+                )
+            }
+            Self::Pyston {
+                python_version: (py_major, py_minor),
+                implementation_version: (impl_major, impl_minor),
+            } => {
+                write!(
+                    f,
+                    "pyston{py_major}{py_minor}-pyston_{impl_major}{impl_minor}"
+                )
+            }
+        }
+    }
+}
+
+impl FromStr for AbiTag {
+    type Err = ParseAbiTagError;
+
+    /// Parse an [`AbiTag`] from a string.
+    #[allow(clippy::cast_possible_truncation)]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
This parsing code is incredibly verbose... It's fairly rote though, and comes with tests. Open to input on how to trim it down. I could perhaps remove all the error-handling...? And just return `None`?